src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTestsMutable.cs

```diff
@@ -0,0 +1,34 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.Tests.Common;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests.RPC
+{
+    /// <summary>
+    /// RPC Integration tests that require their own node(s) for each test because they change node state.
+    /// </summary>
+    public class RPCTestsMutable
+    {
+        [Fact]
+        public void TestRpcGetBlockWithValidHashIsSuccessful()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                var node = builder.CreateStratisPowNode(KnownNetworks.RegTest);
+                builder.StartAll();
+                RPCClient rpcClient = node.CreateRPCClient();
+                node.NotInIBD();
+                node.SetDummyMinerSecret(new BitcoinSecret(new Key(), node.FullNode.Network));
+                node.GenerateStratisWithMiner(2);
```

Comment:
I think we're changing to `TestHelper.MineBlocks` here. I believe there's also a `.WithWallet()` helper method on the node that does some of the boilerplate so you don't have to set the miner secret etc.

Reply:
Hi @mikedennis yes, please wait for my PR to be merged and then you can use the methods as described by @zeptin here 👍

Reply:
Ok thanks. I will take a look!